### ISSUE TITLE
Feature requests for flux-fw setup!

### ISSUE THREAD

**Issue by vsoch** [GoogleCloudPlatform/scientific-computing-examples#56]
Hi @wardharold @wkharold ! I'm thinking it would be nice if we might do the following:

- epose enable-oslogin as a variable (right now hard coded to true)
- provide metadata attributes for the login node and manager (if possibly the user is building a custom setup that warrants that).

What do you think?

**Comment by wkharold** [comment]
> epose enable-oslogin as a variable (right now hard coded to true)

agreed

> provide metadata attributes for the login node and manager (if possibly the user is building a custom setup that warrants that).

I think I understand this one. Do you have an example?

**Comment by vsoch** [comment]
Yeah! The use case is that I'm running a boot script that wants to run custom logic based on if it's the login or manager node. I'm doing a very stupid "well, I know what these are going to be called, so darn it, I'll just check for those names!"

```bash
# We also need the node names - not ideal, but for now they are predictable so it works
node_crio=gffw-compute-a-001
node_containerd=gffw-compute-a-002
node_master=gffw-login-001

# What node is running this?
nodename=$(hostname)

# more stuff here ...

# Run this on the main login node - since it's shared we only need 
# to generate the certs once.
if [[ "$nodename" == *"${node_master}"* ]]; then

    echo "I am the login node ${nodename} going to run the master stuff"
    /bin/bash ./common/cfssl.sh --dir=/home/$user/.config/usernetes --master=${node_master} --node=${node_crio} --node=${node_containerd}

    # 2379/tcp: etcd, 6443/tcp: kube-apiserver
    /bin/bash ./install.sh --wait-init-certs --start=u7s-master-with-etcd.target --cidr=10.0.100.0/24 --publish=0.0.0.0:2379:2379/tcp --publish=0.0.0.0:6443:6443/tcp --cni=flannel --cri=

fi
```
I think what would be more elegant (instead of me hard coding the hostnames) is to be able to retrieve this via the metadata API (akin to how we do for flux-manager) via this section here. https://github.com/GoogleCloudPlatform/scientific-computing-examples/blob/b558d176002aa16fbecff222082cd5d84ce57502/fluxfw-gcp/tf/modules/login/main.tf#L46-L54 

The high level idea is that we shouldn't assume a limited scope of variables that a user might want, and err on the side of exposing more in that little metadata section!

**Comment by vsoch** [comment]
@wkharold I need the OS login pretty badly (it's mucking with the entries in /etc/subuid, etc.) So I'll open a PR for that soon.
